sphinx/builders/html.py

```diff
@@ -400,8 +400,8 @@ def prepare_writing(self, docnames):
 
         favicon = self.config.html_favicon and \
             path.basename(self.config.html_favicon) or ''
-        if favicon and os.path.splitext(favicon)[1] != '.ico':
-            logger.warning('html_favicon is not an .ico file')
+        if favicon and os.path.splitext(favicon)[1] not in ['.ico', '.png']:
+            logger.warning('html_favicon is not an .ico or .png file')
```

Comment:
Now many image formats are allowed to favicon, so there are no meaning to warn about file types here.
https://en.wikipedia.org/wiki/Favicon#Browser_implementation

How about removing this block?